scripts/fetchdep.sh

```diff
@@ -79,11 +76,14 @@ if [[ "$GITHUB_EVENT_NAME" == "merge_group" ]]; then
     clone $deforg $defrepo ${withoutPrefix%%/pr-*}
 fi
 
-# Try the target branch of the push or PR.
-if [ -n "$GITHUB_BASE_REF" ]; then
-    clone $deforg $defrepo $GITHUB_BASE_REF
-elif [ -n "$BUILDKITE_PULL_REQUEST_BASE_BRANCH" ]; then
-    clone $deforg $defrepo $BUILDKITE_PULL_REQUEST_BASE_BRANCH
+# Try the target branch of the push or PR, or the branch that was pushed to
+# (ie. the 'master' branch should use matching 'master' dependencies)
+base_or_branch=$GITHUB_BASE_REF
+if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+    base_or_branch=${GITHUB_REF}
```

Comment:
isn't this what the block above is trying to do with TRY_BRANCH? This implies `$head` isn't set as expected

Reply:
Looks like $head is only set by `getPRInfo()` which is only run for PR builds, so I think that's all just for PRs.